counsel.el

```diff
@@ -3254,6 +3254,36 @@ Note: Duplicate elements of `kill-ring' are always deleted."
  '(("d" counsel-yank-pop-action-remove "delete")
    ("r" counsel-yank-pop-action-rotate "rotate")))
 
+;;** `counsel-evil-registers'
+
+(defcustom counsel-evil-registers-height 5
+  "The `ivy-height' of `counsel-evil-registers'."
+  :group 'ivy
+  :type 'integer)
+
+(defun counsel-evil-registers ()
+  "Ivy replacement for `evil-show-registers'."
+  (interactive)
+  (let ((ivy-format-function #'counsel--yank-pop-format-function)
+        (ivy-height counsel-evil-registers-height))
+    (ivy-read "evil-registers: "
+              (cl-loop for (key . val) in (evil-register-list)
+                       collect
+                       (format "[%s]: %s" (char-to-string key)
+                               (or (and val (stringp val) val) "")))
+              :require-match t
+              :action #'counsel-evil-registers-action
+              :caller 'counsel-evil-registers)))
+
+(defun counsel-evil-registers-action (s)
+  "Paste contents of S, trimming the register part.
+
+S will be of the form \"[register]: content\"."
+  (with-ivy-window
+    (setq last-command 'yank)
```

Comment:
Why is setting `last-command` necessary?

Reply:
Honestly, dunno. Copied it from counsel-yank-pop. I could remove it if we don't think it's necessary.

Reply:
In that case you can remove it - `counsel-yank-pop` needs to set it before calling `yank-pop`.